analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -125,8 +125,11 @@ public bool HasMatchingScope(DiagnosticDescriptor descriptor)
             descriptor.CustomTags.Contains(tag);
     }
 
-    private bool ShouldAnalyzeFile(SonarLintXmlReader sonarLintXml, string filePath) =>
-        ProjectConfiguration().ProjectType != ProjectType.Unknown // Not SonarLint context, NuGet or Scanner <= 5.0
+    private bool FileIsIncludedInAnalysis(SonarLintXmlReader sonarLintXml, string filePath) =>
+        // If ProjectType != 'Unknown' we are in S4NET context and all files are analyzed.
+        // If ProjectType == 'Unknown' then we are in SonarLint or NuGet context and
+        // we need to check if the file has been excluded from analysis through SonarLint.xml.
+        ProjectConfiguration().ProjectType != ProjectType.Unknown
```

Comment:
The comment is now clear. It's this `!=` that makes it still hard. 
Experiment:
What about making this method `IsExcluded(sonarLintXml, filePath)`?
The implementation would be something like
```
ProjectType ==  Unknown
&& ... is var cache
&& !.... IsFileIncluded ...
```
and it could also play nicely with `ShouldAnalyzeFile` method

Reply:
This method is the old `ShouldAnalyzeFile` (see [here](https://github.com/SonarSource/sonar-dotnet/blob/feature/SL-exclusions-inclusions/analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs#L128)).

IMO I prefer to leave the name as `IsExcluded` - it's clear and it goes well with "IsUnchanged(tree)" in the same condition.